src/transactions/test/SorobanTxTestUtils.cpp

```diff
@@ -1097,28 +1106,24 @@ SorobanTest::isTxValid(TransactionTestFramePtr tx)
     return ret;
 }
 
-bool
+TransactionResult
```

Comment:
There's an interface inconsistency here with returning a `TransactionResult` object while taking in a `TransactionTestFramePtr` parameter. The idea behind `TransactionTestFramePtr` is that the `TransactionResult` is always bundled with the tx object, allowing for calls like `tx->getResult`. The issue is, here we call `ledgerClose({tx})`. This will convert `{tx}` into a XDR TxSet object, and since we're calling the prod `ledgerClose` path, the transaction inside the XDR TxSet get converted to non-test `TransactionFrame` objects. The result is, the `TransactionResult` returned by `invokeTx` is different than the `TransactionResult` stored inside the test `tx` object. This seems like a footgun, as many tx tests rely on `TransactionTestFrame::getResult`. Sorry, this was a short sight on my end and something I ran into on my test refactors as well.

What I've been doing is using the result set from `closeLedger` to set the cached `TransactionResult` inside `TransactionTestFrame` ([see `applyTx`in my branch](https://github.com/SirTyson/stellar-core/blob/cd7f3100208eb13606ada0018199f032796a23fb/src/test/TxTests.cpp#L388)). I think it's going to be kinda hacky either way given the nature of `TransactionTestFrame` (my bad), but I prefer keeping `TransactionTestFrame->getResult` correct over returning the result to be consistent with other test interfaces, open to suggestions though.

Reply:
We only use test frame (anti-)pattern for the legacy tests, here it's easy enough to just use the proper tx frame. The same goes for the loadgen (which happens to use this function), so I just got rid of test frames for both.